### DISCUSSION TITLE
Filter GraphQL with Custom Fields

### DISCUSSION THREAD

**Discussion by AbdelrahmanKabel** [OrchardCMS/OrchardCore#12630] [Q&A] [answered]
Hello,

I have created a Content Type "ContactUS" with custom fields "name, email", now I want to query the items with GraphQL filterd by my custom fields but I can only filter using the predefined properties.

This is my query:
query MyQuery {
  contactUS(where: {**email**: "value"}) {
    displayText
    name
    email
  }
}

But if I changed the filtration to be by **displayText** it works like below.
query MyQuery {
  contactUS(where: {**displayText**: "value"}) {
    displayText
    name
    email
  }
}

How can I filter in GraphQL using my custom fields?

**Comment by Piedone** [comment, marked as answer]
This is possible but not without development. You'd need to implement a `WhereInputObjectGraphType` for the field to be able to filter on it (see the OC source for samples, as well as [this one](https://github.com/Lombiq/Orchard-Training-Demo-Module/blob/dev/GraphQL/Services/PersonPartWhereInputObjectGraphType.cs)).

For an alternative with SQL queries, see [Content Field Indexing](https://docs.orchardcore.net/en/latest/docs/reference/modules/SQLIndexing/#content-fields-indexing).

**Reply by Piedone** [reply]
I opened a related feature request: https://github.com/OrchardCMS/OrchardCore/issues/12781

**Reply by AbdelrahmanKabel** [reply]
Thank you @Piedone It's really a helpful answer. but now I'm trying to do pagination using **skip** and **first** I'm missing here the **total count** in order to have the number of pages. Is there is a way to return the total count?

**Reply by Piedone** [reply]
That seems like a new question; please open a new discussion for that, and accept the answer here.
